connectivity/tests/common.go

```diff
@@ -48,3 +54,79 @@ func hasAllLabels(labelsContainer labelsContainer, filters map[string]string) bo
 	}
 	return true
 }
+
+type retryCondition struct {
+	podLabels map[string]string
+	all       bool
+	destPort  uint32
+	destIP    string
+}
+
+// CurlOptions returns curl retry option or empty slice depending on retry conditions
+func (rc *retryCondition) CurlOptions(peer check.TestPeer, ipFam check.IPFamily, pod check.Pod, params check.Parameters) []string {
+	if params.Retry == 0 {
+		return []string{}
+	}
+	if !rc.all && rc.destIP == "" && rc.destPort == 0 {
```

Comment:
Shouldn't we extend this to also consider empty pod labels?

```suggestion
	if !rc.all && rc.destIP == "" && rc.destPort == 0 && len(rc.podLabels) == 0 {
```

Reply:
Pool labels are only used with other conditions, in this case only with destPort.